File: Algorithms/1346. Check If N and Its Double Exist/javascript/solution.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function (arr) {
    const seen = new Set();
    for (const value of arr) {
        if (seen.has(value * 2) || seen.has(value / 2)) {
            return true;
        }
        seen.add(value);
    }
    return false;
};

File: Algorithms/1346. Check If N and Its Double Exist/golang/solution.go
func checkIfExist(arr []int) bool {
	seen := make(map[float64]bool)
	for _, v := range arr {
		V := float64(v)
		if seen[V*2.0] || seen[V/2.0] {
			return true
		}
		seen[V] = true
	}
	return false
}